duecredit/io.py

```diff
@@ -132,12 +134,12 @@ def dump(self, tags=None):
                 # description so must be groupped accordingly. For now just simply listing them
                 # all separately
                 for citation in citations:
-                    refnr = len(citations_ordered) + 1
+                    refnr += 1
                     self.fd.write('  - {0} ({1}) [{2}]\n'.format(
                         citation.path,
                         citation.description,
                         refnr))
-                    citations_ordered.extend(citations)
+                citations_ordered.extend(citations)
```

Comment:
if would still need to be reworked since e.g. if now the same bib reference used for multiple packages (and or subpackages) it will get listed twice ... it is not just a fix in logic here -- it is how to do it altogether.  We need to some abstraction on top of all entries to be referenced, and that one is the one to provide their numbers and at the end "render" in some form.